vortex-array/src/context.rs

```diff
@@ -8,54 +12,129 @@ use crate::arrays::{
 use crate::encoding::Encoding;
 use crate::vtable::VTableRef;
 
-/// A mapping between an encoding's ID to an [`VTableRef`], used to have a shared view of all available encoding schemes.
-#[derive(Debug, Clone)]
-pub struct Context {
-    encodings: HashMap<u16, VTableRef>,
+/// A collection of array encodings.
+// TODO(ngates): it feels weird that this has interior mutability. I think maybe it shouldn't.
```

Comment:
This is the thing that tripped me up a bit when reading this pr. I was initially confused how we make sure we collect all the encodings but then I noticed we pass the context through array node serialization. Having said that it's probably hard to write this in a way where it doesn't have interior mutability.